spec/controllers/concerns/idv/document_capture_concern_spec.rb

```diff
@@ -186,26 +215,66 @@ def show
       stored_result = EncryptedRedisStructStorage.load(id, type: DocumentCaptureSessionResult)
       allow(controller).to receive(:stored_result).and_return(stored_result)
       allow(controller).to receive(:document_capture_session).and_return(document_capture_session)
+      allow(document_capture_session).to receive(:passport_requested?).and_return(false)
+      allow(document_capture_session).to receive(:doc_auth_vendor)
+        .and_return(Idp::Constants::Vendors::MOCK)
+      allow(IdentityConfig.store).to receive(:doc_auth_passports_enabled).and_return(true)
     end
 
-    context 'when document is a state ID' do
+    context 'when passport not requested' do
       before do
-        allow(controller).to receive(:id_type).and_return('state_id')
+        allow(document_capture_session).to receive(:passport_requested?).and_return(false)
       end
 
-      it 'returns true regardless of mrz_status' do
+      it 'returns true' do
         expect(controller.mrz_requirement_met?).to eq(true)
       end
     end
 
+    context 'when passport requested but state ID submitted' do
+      before do
+        allow(document_capture_session).to receive(:passport_requested?).and_return(true)
+      end
+
+      it 'returns false' do
+        expect(controller.mrz_requirement_met?).to eq(false)
+      end
+    end
+
     context 'when document is a passport' do
       before do
-        allow(controller).to receive(:id_type).and_return('passport')
+        allow(document_capture_session).to receive(:passport_requested?).and_return(true)
       end
 
       context 'when mrz_status is pass' do
         let(:mrz_status) { :pass }
 
+        before do
+          # Update the stored result to include valid passport PII
+          id = SecureRandom.hex
+          result = DocumentCaptureSessionResult.new(
+            id:,
+            success: true,
+            doc_auth_success: true,
+            selfie_status: :not_processed,
+            mrz_status:,
+            pii: {
+              mrz: "P<USADOE<<JOHN<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<\n" \
+                 "123456789<USA7006100M2501017<<<<<<<<<<<<<<06\n",
+              document_number: '123456789',
+              passport_expiration: '2030-01-01',
+              issuing_country_code: 'USA',
+              nationality_code: 'USA',
+              dob: '1970-06-10',
+              id_doc_type: 'passport',
+            },
+            attention_with_barcode: false,
+          )
+          EncryptedRedisStructStorage.store(result)
+          stored_result = EncryptedRedisStructStorage.load(id, type: DocumentCaptureSessionResult)
+          allow(controller).to receive(:stored_result).and_return(stored_result)
```

Comment:
And another argument for defining lets/allows at the top - here you are defining the result, storing it, loading it and allowing its return... _after_ defining/storing/loading/allowing in the before from lines 205-216 :-P